File: src/ipc/health.ts
// Typed wrappers for the `health` domain (W1 stub). Establishes the per-domain
// wrapper-module pattern every later item follows: one module per backend
// domain, each thin function calling `invoke` with the command name + args.

import { invoke } from "./invoke";

/** Round-trips a liveness check through the backend; returns its reply string. */
export function ping(): Promise<string> {
  return invoke<string>("ping");
}

File: src-tauri/src/commands/health.rs
//! Health domain (W1 stub). A trivial command the frontend round-trips to prove
//! the IPC seam end-to-end. Later items add real domain adapters as sibling
//! files under `commands/`; this one stays as the canonical minimal example.

use crate::error::AppResult;

/// Liveness check. Returns a fixed reply the frontend renders to prove the
/// invoke pipeline (TS wrapper → Tauri → Rust → typed return) works.
#[tauri::command]
pub async fn ping() -> AppResult<String> {
    Ok("pong".to_string())
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn ping_returns_pong() {
        // Drive the async command on Tauri's runtime so the test needs no
        // direct tokio dependency.
        let reply = tauri::async_runtime::block_on(ping()).unwrap();
        assert_eq!(reply, "pong");
    }
}
